receiver/googlecloudmonitoringreceiver/receiver_test.go

```diff
@@ -0,0 +1,872 @@
+func TestInitializeClient_Success(t *testing.T) {
+	ctx := context.Background()
+
+	// Set up fake credentials for testing
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "./../googlecloudspannerreceiver/testdata/serviceAccount.json")
+	defer os.Unsetenv("GOOGLE_APPLICATION_CREDENTIALS")
+
+	logger := zap.NewNop()
+	cfg := &Config{
+		ProjectID:   "fake-project",
+		MetricsList: []MetricConfig{{MetricName: "custom.googleapis.com/fake_metric"}},
+		ControllerConfig: scraperhelper.ControllerConfig{
+			CollectionInterval: 1 * time.Minute,
+		},
+	}
+
+	receiver := newGoogleCloudMonitoringReceiver(cfg, logger)
+	err := receiver.initializeClient(ctx)
+	assert.NoError(t, err)
+}
+
+func TestInitializeClient_Failure(t *testing.T) {
+	ctx := context.Background()
+
+	logger := zap.NewNop()
+	cfg := &Config{
+		ProjectID:   "fake-project",
+		MetricsList: []MetricConfig{{MetricName: "custom.googleapis.com/fake_metric"}},
+		ControllerConfig: scraperhelper.ControllerConfig{
+			CollectionInterval: 1 * time.Minute,
+		},
+	}
+
+	receiver := newGoogleCloudMonitoringReceiver(cfg, logger)
+	err := receiver.initializeClient(ctx)
+	assert.Error(t, err)
+}
+
+func TestStart_Success(t *testing.T) {
+	ctx := context.Background()
+	testServer, err := NewFakeMetricTestServer()
+	require.NoError(t, err)
+	go testServer.srv.Serve(testServer.lis)
+	defer testServer.srv.Stop()
+
+	clientOpts := []option.ClientOption{
+		option.WithEndpoint(testServer.Endpoint),
+		option.WithoutAuthentication(),
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+	}
+
+	logger := zap.NewNop()
+	cfg := &Config{
+		ProjectID:   "fake-project",
+		MetricsList: []MetricConfig{{MetricName: "custom.googleapis.com/fake_metric"}},
+		ControllerConfig: scraperhelper.ControllerConfig{
+			CollectionInterval: 1 * time.Minute,
+		},
+	}
+
+	receiver := newGoogleCloudMonitoringReceiver(cfg, logger)
+	receiver.client, err = monitoring.NewMetricClient(ctx, clientOpts...)
+	require.NoError(t, err)
+
+	// Test the Start function
+	err = receiver.Start(ctx, nil)
+	assert.NoError(t, err)
+
+	// Verify that the receiver is properly initialized
+	assert.NotNil(t, receiver.client)
+	assert.NotNil(t, receiver.metricDescriptors)
+}
+
+func TestStart_Failure_NoClient(t *testing.T) {
+	ctx := context.Background()
+	logger := zap.NewNop()
+	cfg := &Config{
+		ProjectID:   "fake-project",
+		MetricsList: []MetricConfig{{MetricName: "custom.googleapis.com/fake_metric"}},
+		ControllerConfig: scraperhelper.ControllerConfig{
+			CollectionInterval: 1 * time.Minute,
+		},
+	}
+
+	receiver := newGoogleCloudMonitoringReceiver(cfg, logger)
+	// Don't initialize the client to test failure case
+
+	// Test the Start function
+	err := receiver.Start(ctx, nil)
+	errMsg := fmt.Errorf("%s", "failed to find default credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information")
```

Comment:
lets avoid checking for specific error strings, as these may change over time.  Can we maybe just check for "credentials", or something less likely to change?